iis-wf/iis-wf-ingest-pmc/src/test/java/eu/dnetlib/iis/wf/ingest/pmc/metadata/RefListXmlHandlerTest.java

```diff
@@ -0,0 +1,119 @@
+
+import eu.dnetlib.iis.ingest.pmc.metadata.schemas.ExtractedDocumentMetadata;
+import eu.dnetlib.iis.ingest.pmc.metadata.schemas.ReferenceBasicMetadata;
+import eu.dnetlib.iis.ingest.pmc.metadata.schemas.ReferenceMetadata;
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.xml.sax.InputSource;
+
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+import java.nio.charset.StandardCharsets;
+import java.util.AbstractMap;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class RefListXmlHandlerTest {
+
+    private SAXParser saxParser;
+    private ExtractedDocumentMetadata.Builder builder;
+    private RefListXmlHandler refListXmlHandler;
+
+    @BeforeEach
+    public void init() throws Exception {
+        SAXParserFactory saxFactory = SAXParserFactory.newInstance();
+        saxFactory.setValidating(false);
+        saxParser = saxFactory.newSAXParser();
+        builder = ExtractedDocumentMetadata.newBuilder();
+        builder.setId("id");
+        builder.setEntityType("entity-type");
+        builder.setText("text");
+        refListXmlHandler = new RefListXmlHandler(builder);
+    }
+
+    @Test
+    @DisplayName("References with 'nlm-citation' element are parsed")
+    public void givenRefListXmlHandler_whenRefWithNlmCitationIsParsed_thenReferenceMetadataWithAllDataIsExtracted() throws Exception {
+        String input = "<ref content-type=\"article\" id=\"B1\"><label>1</label><nlm-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name><surname>McCandlish</surname><given-names>R.</given-names></name><name><surname>Bowler</surname><given-names>U.</given-names></name><name><surname>van Asten</surname><given-names>H.</given-names></name><name><surname>Berridge</surname><given-names>G.</given-names></name><name><surname>Winter</surname><given-names>C.</given-names></name><name><surname>Sames</surname><given-names>L.</given-names></name><name><surname>Garcia</surname><given-names>J.</given-names></name><name><surname>Renfrew</surname><given-names>M.</given-names></name><name><surname>Elbourne</surname><given-names>D.</given-names></name></person-group><article-title>A randomised controlled trial of care of the perineum during second stage of normal labour</article-title><source><italic>British Journal of Obstetrics and Gynaecology</italic></source><year>1998</year><volume>105</volume><issue>12</issue><fpage>1262</fpage><lpage>1272</lpage><pub-id pub-id-type=\"doi\">10.1111/j.1471-0528.1998.tb10004.x</pub-id><pub-id pub-id-type=\"other\">2-s2.0-0032423003</pub-id></nlm-citation></ref>";
```

Comment:
I can almost hear this variable screaming for having its content defined within an external classpath resource ;) Especially seeing the line below we read it into an `InputSource` which could be easily done relying on an XML file stored in classpath.

Reply:
It was my initial idea :) But this package has a lot of test resources added. This is the `data` folder content:
```text
iis-wf/iis-wf-ingest-pmc/src/test/resources/eu/dnetlib/iis/wf/ingest/pmc/metadata/data/
├── articlemeta
│   ├── authors_with_aff_in_contrib_group.xml
│   ├── authors_with_aff_in_contrib.xml
│   ├── authors_with_aff_refs.xml
│   └── authors_with_encoded_characters.xml
├── document_gapped.xml
├── document_invalid.xml
├── document_jats10_nested_in_oai.xml
├── document_jats10.xml
├── document_jats23_nested_in_oai.xml
├── document.json
├── document_with_affiliations.xml
├── od_______908__0451fa1ded79a63729296731e53335c0.xml
├── od_______908__0452195ccf851072fd097fc49bfbb9da.xml
├── od_______908__365a50343d53774f68fa13800349d372.xml
└── single-ref-document.xml
``` 
I thought it will be better to not add another file with only a part of PMC XML. That's why I've put it as a variable in test code.